### DISCUSSION TITLE
Prune operation in Borg 2

### DISCUSSION THREAD

**Discussion by stevezemlicka** [borgbackup/borg#7922] [Q&A] [answered]
I'm new to the borgbackup project.  I am basing my script off of the following resource:
https://borgbackup.readthedocs.io/en/2.0.0a2/quickstart.html#a-step-by-step-example

I was receiving errors during prune due to --prefix not being a valid option so it seems that, maybe, this is a relic of Borg 1.  Looking at the help, it seems that maybe -a should be used for pattern matching but I'm not really certain about the syntax.  My intention is to use it similar to --prefix because I want the prune command to only affect archives starting with a pattern (or containing the pattern since the archives are named "pattern-{now}".  For example, would a borg2 prune operation only affect archives with the text 'pattern' in the name with the following option:
`-a 'pattern'`

Another random question.  I cannot seem to get borg2 to run without adding a file.  I have 2 datasets that I'm backing up and both of them exhibit this symptom with different files.  If I remove the file in question, it just changes to a different (already in existence and unchanged) file.  For example, running the backup twice in relatively quick succession (fwiw, the checksum of the file "/mnt/general/HEFC/Video Projects/2023-11-05/thumb.jpg" is unchanged throughout both of these runs):
```
@borg01:~$ ./borg2-full-backup.sh 

Wed Nov  8 04:34:56 AM CST 2023 Starting backup

Creating archive at "/mnt/onsite-backups/borg2-repo"
A /mnt/general/HEFC/Video Projects/2023-11-05/thumb.jpg
Repository: /mnt/onsite-backups/borg2-repo
Archive name: general-2023-11-08T04:34:56
Archive fingerprint: 568b52c0b307fa57dfb7d8909ce97ff942761464d5f0d61ed22692d11c6a22eb
Time (start): Wed, 2023-11-08 04:34:57 -0600
Time (end):   Wed, 2023-11-08 04:36:09 -0600
Duration: 1 minutes 11.17 seconds
Number of files: 60171
Original size: 85.21 GB
Deduplicated size: 2.16 kB
Time spent in hashing: 0.45 seconds
Time spent in chunking: 0.00 seconds
Added files: 1
Unchanged files: 60170
Modified files: 0
Error files: 0
Files changed while reading: 0
Bytes read from remote: 0
Bytes sent to remote: 0
terminating with success status, rc 0
```

```
@borg01:~$ ./borg2-full-backup.sh 

Wed Nov  8 04:36:47 AM CST 2023 Starting backup

Creating archive at "/mnt/onsite-backups/borg2-repo"
A /mnt/general/HEFC/Video Projects/2023-11-05/thumb.jpg
Repository: /mnt/onsite-backups/borg2-repo
Archive name: general-2023-11-08T04:36:47
Archive fingerprint: 859815160fc8584596f4c3f0ce1c27276830e3733fb9bf03d2dde4e61f3d9c6d
Time (start): Wed, 2023-11-08 04:36:49 -0600
Time (end):   Wed, 2023-11-08 04:37:57 -0600
Duration: 1 minutes 8.03 seconds
Number of files: 60171
Original size: 85.21 GB
Deduplicated size: 2.16 kB
Time spent in hashing: 0.44 seconds
Time spent in chunking: 0.00 seconds
Added files: 1
Unchanged files: 60170
Modified files: 0
Error files: 0
Files changed while reading: 0
Bytes read from remote: 0
Bytes sent to remote: 0
terminating with success status, rc 0
```
It's not really causing a problem per se, but it seems inaccurate and is definitely triggering my ocd lol.

**Comment by ThomasWaldmann** [comment, marked as answer]
The latest beta is b7, but you are referring to a2 docs.

b7 docs: https://borgbackup.readthedocs.io/en/2.0.0b7/quickstart.html

They show correct usage with `--match-archives` now.

See the FAQ about adding that unchanged file:

https://borgbackup.readthedocs.io/en/2.0.0b7/faq.html#i-am-seeing-a-added-status-for-an-unchanged-file

**Comment by stevezemlicka** [comment]
Awesome, thank you.  That worked perfectly and clarifies things.

I just now saw the drop-down in the bottom right corner to select the different versions on that docs page.  I wonder if that feature would be more intuitive/obvious if it were in the upper left corner of the page (like next to the logo/version).

**Reply by ThomasWaldmann** [reply]
Filed https://github.com/borgbackup/borgbackup.github.io/issues/105 .
